Disable caching on GitHub-host MacOS X86_64 runners

GMNGeoffrey opened the pull request:
The GitHub Actions cache is utterly broken. See
https://github.com/actions/cache/issues/1115 and the many commits in
this PR trying to get it working.

Part of https://github.com/openxla/iree/issues/12727

GMNGeoffrey committed:
Dramatically limit the size of GitHub Actions Mac Cache

GMNGeoffrey committed:
REVERT ME: Run mac CI on presubmit

ScottTodd commented on .github/workflows/ci.yml:
```diff
@@ -343,7 +343,11 @@ jobs:
+          # https://github.com/actions/cache/issues/1115. Hopefully this is
+          # still large enough to be somewhat useful.
+          CCACHE_MAXSIZE: 1.5G
           CCACHE_COMPRESSLEVEL: 5
```
I'm not optimistic about the cache being as useful as we'd like with the reduced size. Some cache is hopefully better than no cache though.

ScottTodd replied:
Do you want to test this PR with cache writes enabled? The presubmit run with existing cache entries downloaded ~2795 MB: https://github.com/openxla/iree/actions/runs/4601599229/jobs/8129656448?pr=12905#step:7:247

GMNGeoffrey replied:
Yeah sure. I wasn't sure if it would actually work since it's coming from a fork, but we'll see

GMNGeoffrey replied:
So the good news is that fetching the cache only takes a minute. The bad news is that the build with this cache is not any faster than an uncached build. Ugh this is so annoying

GMNGeoffrey committed:
REVERT ME: Write mac cache on presubmit and turn off other jobs

GMNGeoffrey committed:
REVERT ME: Actually upload cache

GMNGeoffrey committed:
Try initially larger cache with recompression

GMNGeoffrey committed:
Specify CCACHE_DIR for all steps

GMNGeoffrey committed:
Don't clear CCACHE_DIR accidentally...

ScottTodd commented on .github/workflows/ci.yml:
```diff
@@ -325,36 +325,36 @@ jobs:
-          path: ${{ github.workspace }}/.ccache
-          key: ccache_all_macos_${{ github.sha }}
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ env.CCACHE_GH_CACHE_KEY }}_${{ github.sha }}
```
https://github.com/openxla/iree/actions/runs/4623460718/jobs/8178728862?pr=12905#step:11:18

`Failed to save: Unable to reserve cache with key ccache_all_macos_cb7ce722eb10c896195ff203c9be77005c7457d8, another job may be creating this cache. More details: Cache already exists. Scope: refs/pull/12905/merge, Key: ccache_all_macos_cb7ce722eb10c896195ff203c9be77005c7457d8, Version: 9f9bec005f64399df969515b921f6f040690a403ecdee39a26aec50792275dbe`

GMNGeoffrey replied:
Oh ffs. I guess the cache key has to be unique per *run*? That's so stupid

GMNGeoffrey replied:
Or maybe because the cache is actually identical

GMNGeoffrey replied:
Ah looks like the issue comes from https://github.com/actions/toolkit/issues/658. Seems cache keys really do have to be unique. That scuttles our plans to cache based on LLVM commit. Wow that is so stupid...

GMNGeoffrey committed:
Log ccache stats before running. Useful for full directory caches

GMNGeoffrey committed:
Drop the max cache size

We haven't had issues in these test runs, but this isn't actually hitting the limit.

GMNGeoffrey committed:
Poke CI since reruns upset the caching

GMNGeoffrey committed:
Turn off caching on mac entirely
